app/models/manager_refresh/save_collection/saver/concurrent_safe_batch.rb

```diff
@@ -99,11 +99,21 @@ def save!(inventory_collection, association)
       end
 
       def destroy_records(records)
-        # TODO(lsmola) we need at least batch disconnect. Batch destroy won't be probably possible because of the
-        # :dependent => :destroy.
-        ActiveRecord::Base.transaction do
-          records.each do |record|
-            delete_record!(inventory_collection, record)
+        return false unless inventory_collection.delete_allowed?
+        return if records.blank?
+
+        # Is the delete_method rails standard deleting method?
+        rails_delete = %i(destroy delete).include?(inventory_collection.delete_method)
+        if !rails_delete && inventory_collection.model_class.respond_to?(inventory_collection.delete_method)
```

Comment:
Perhaps InventoryCollection having `:batch_delete_method` param, instead of magic implication of having class method with same name?